src/RoslynPad/DocumentViewModel.cs

```diff
@@ -19,16 +19,76 @@ internal class DocumentViewModel : NotificationObject
         private DocumentViewModel(MainViewModel mainViewModel)
         {
             MainViewModel = mainViewModel;
-            var defaultPath = GetDefaultPath();
-            Directory.CreateDirectory(defaultPath);
-            Path = defaultPath;
+            Path = GetUserDocumentPath();
+            Directory.CreateDirectory(Path);
             IsFolder = true;
         }
 
-        public string Path { get; }
+        public string Path { get; internal set; }
+
         public MainViewModel MainViewModel { get; }
         public bool IsFolder { get; }
 
+        internal static string GetUserDocumentPath()
+        {
+            var userDefinedPath = Properties.Settings.Default.DocumentPath;
+            return !string.IsNullOrEmpty(userDefinedPath) && System.IO.Directory.Exists(userDefinedPath)
+                ? userDefinedPath
+                : GetDefaultDocumentPath();
+        }
+
+        private static string GetDefaultDocumentPath()
+        {
+            return System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "RoslynPad");
+        }
+
+        public async void EditUserDocumentPath()
+        {
+            // TODO: Review expected behavior. 
+            // Currently rebinding documents pane.
+            // Dialog requesting restart? Auto-restart app? Rebind documents pane? What happens to open documents?
+
+            var dialog = new System.Windows.Forms.FolderBrowserDialog()
```

Comment:
We'll use the `Avalon.Windows.Dialogs.FolderBrowserDialog`, which doesn't require win forms reference (it's a library of mine already referenced in RP)